File: javascript/문자열압축.js
function solution(s) {
  var answer = 0;
  let str, cnt;
  let maxStr, maxCnt;
  let maxLen = 0;
  for (let i = 1; i <= parseInt(s.length / 2); i++) {
    let tmp = {};
    let prev = s.substr(0, i);
    let curLen = 0;
    cnt = 1;
    for (let iter = i; iter < s.length; iter += i) {
      const cur = s.substr(0 + iter, i);
      if (prev === cur) {
        cnt++;
      } else {
        if (cnt >= 2) {
          curLen += prev.length * cnt - (String(cnt).length + prev.length);
        }
        prev = cur;
        cnt = 1;
      }
    }
    if (cnt >= 2) {
      curLen += prev.length * cnt - (String(cnt).length + prev.length);
    }

    console.log(tmp);
    maxLen = Math.max(maxLen, curLen);
    console.log(maxLen);
  }
  //   console.log(str);
  console.log(maxStr, maxCnt);
  return s.length - maxLen;
}
let c = 'xababcdcdababcdcd';
console.log(solution(c));

File: java/문자열압축.java
class Solution {
    public static int getLen(int size , String s){
        int ret=0;
        int cnt=0;
        String tmp="";
        for(int i=0;i+size<=s.length();i+=size){
            String cur = s.substring(i, i+size);
            if(tmp.equals(cur)){
                cnt++;
            }else{
                ret+=(tmp.length());
                if(cnt>1){
                    ret+=Integer.toString(cnt).length();
                }
                tmp = cur;
                cnt=1;
            }
        }
        ret+=(tmp.length());
        if(cnt>1){
            ret+=Integer.toString(cnt).length();
        }
        ret+=(s.length()%size);
        return ret;
    }
    public int solution(String s) {
        int len = s.length();
        int answer = len;
        for(int i=1;i<=len/2+1;i++){
            answer = Math.min(answer , getLen(i,s));
        }
        return answer;
    }
}